### PR TITLE
wownero: init at 0.4.0.0

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

Add the cryptocurrency Wownero to nixpkgs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by fuwa0529** [comment]
@jwinterm for review

**Comment by fuwa0529** [comment]
updated the PR to point to the latest release

**Comment by fuwa0529** [comment]
updated again to point to the latest release

File: pkgs/applications/altcoins/wownero.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+assert stdenv.isDarwin -> IOKit != null;
+
+with stdenv.lib;
+
+stdenv.mkDerivation rec {
+  name = "wownero-${version}";
+
+  version = "0.3.1.3";
+  src = fetchgit {
```
Comment (kalbasit) [inline pkgs/applications/altcoins/wownero.nix]: please use `fetchFromGitHub` instead of `fetchgit`

File: pkgs/applications/altcoins/wownero.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+      Wownero’s emission is capped and supply is finite.
+
+      Wownero is a fairly launched coin with no premine.
+
+      It’s not a fork of another blockchain. With its own genesis block there is no degradation of privacy caused by ring signatures using different participants for the same transaction outputs.
+
+      Unlike opposing forks.
+    '';
+    homepage    = https://wownero.org/;
```
Comment (kalbasit) [inline pkgs/applications/altcoins/wownero.nix]: They don't respond to https, http://wownero.org/ works though.

**Review by kalbasit — CHANGES_REQUESTED** [review]

File: pkgs/applications/altcoins/wownero.nix (hunk 1 above)
Comment (fuwa0529) [reply pkgs/applications/altcoins/wownero.nix]: Thanks @kalbasit, i needed the `fetchSubmodules = true;` part from fetchgit, is it support in fetchFromGithub now?

File: pkgs/applications/altcoins/wownero.nix (hunk 2 above)
Comment (fuwa0529) [reply pkgs/applications/altcoins/wownero.nix]: Thanks, reported in the IRC, but it should be fixed soon.

File: pkgs/applications/altcoins/wownero.nix (hunk 1 above)
Comment (kalbasit) [reply pkgs/applications/altcoins/wownero.nix]: Yes it's supported, see the source for fetchFromGitHub [here](https://github.com/NixOS/nixpkgs/blob/05a75a40e6954a4657be353b8358ba4e3686dbf1/pkgs/top-level/all-packages.nix#L260-L291)

Comment (fuwa0529) [reply pkgs/applications/altcoins/wownero.nix]: Thanks! updated.

File: pkgs/applications/altcoins/wownero.nix (hunk 2 above)
Comment (fuwa0529) [reply pkgs/applications/altcoins/wownero.nix]: url updated to use plain http.

File: pkgs/applications/altcoins/wownero.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    longDescription = ''
+      Wownero is a fork of the cryptocurrency Monero with primary alterations.
+
+      Wownero’s emission is capped and supply is finite.
+
+      Wownero is a fairly launched coin with no premine.
+
+      It’s not a fork of another blockchain. With its own genesis block there is no degradation of privacy caused by ring signatures using different participants for the same transaction outputs.
+
+      Unlike opposing forks.
```
Comment (Infinisil) [inline pkgs/applications/altcoins/wownero.nix]: Can you fix the formatting here? Only needs newlines to keep the length under 80 columns, no other newlines needed.

File: pkgs/applications/altcoins/wownero.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  cmakeFlags = [
+    "-DReadline_ROOT_DIR=${readline.dev}"
+    "-DMANUAL_SUBMODULES=ON"
+  ] ++ optional stdenv.isDarwin "-DBoost_USE_MULTITHREADED=OFF";
+
+  hardeningDisable = [ "fortify" ];
+
+  meta = {
+    description = "Many coins, such privacy. Wow!";
```
Comment (Infinisil) [inline pkgs/applications/altcoins/wownero.nix]: This isn't a very descriptive description.. Can you put something better here?

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -1511,6 +1511,11 @@
     github = "fuuzetsu";
     name = "Mateusz Kowalczyk";
   };
+  fuwa = {
+    email = "echowss@gmail.com";
+    github = "fuwa0529";
+    name = "Haruka Akiyama";
+  };
```
Comment (Infinisil) [inline maintainers/maintainer-list.nix]: Put this change into a separate commit in this PR.

**Review by infinisil — CHANGES_REQUESTED** [review]

File: pkgs/applications/altcoins/wownero.nix (hunk 3 above)
Comment (fuwa0529) [reply pkgs/applications/altcoins/wownero.nix]: Thanks for the information, fixed.

File: pkgs/applications/altcoins/wownero.nix (hunk 4 above)
Comment (fuwa0529) [reply pkgs/applications/altcoins/wownero.nix]: I updated the description to use the first line of the long description, which happens to be the summary of it.

File: maintainers/maintainer-list.nix (hunk 5 above)
Comment (fuwa0529) [reply maintainers/maintainer-list.nix]: Done

**Comment by Infinisil** [comment]
@GrahamcOfBorg build altcoins.wownero

**Review by infinisil — APPROVED** [review]
